functions/Grant-DbaAgPermission.ps1

```diff
@@ -121,30 +121,28 @@ function Grant-DbaAgPermission {
         }
 
         foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
+            } catch {
+                Stop-Function -Message "Error occurred while establishing connection to $instance" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
             if ($perm -contains "CreateAnyDatabase") {
-                try {
-                    $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
-                } catch {
-                    Stop-Function -Message "Error occurred while establishing connection to $instance" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
-                }
-
                 foreach ($ag in $AvailabilityGroup) {
                     try {
                         $server.Query("ALTER AVAILABILITY GROUP $ag GRANT CREATE ANY DATABASE")
                     } catch {
-                        Stop-Function -Message "Failure" -ErrorRecord $_ -Target $instance
-                        return
+                        Stop-Function -Message "Failure" -ErrorRecord $_ -Target $instance -Continue
                     }
                 }
-            } elseif ($Login) {
-                $InputObject += Get-DbaLogin -SqlInstance $instance -SqlCredential $SqlCredential -Login $Login
+            }
+            if ($Login) {
+                $InputObject += Get-DbaLogin -SqlInstance $server -SqlCredential $SqlCredential -Login $Login
                 foreach ($account in $Login) {
                     if ($account -notin $InputObject.Name) {
                         try {
                             $InputObject += New-DbaLogin -SqlInstance $server -Login $account -EnableException
                         } catch {
-                            Stop-Function -Message "Failure" -ErrorRecord $_ -Target $instance
-                            return
+                            Stop-Function -Message "Failure" -ErrorRecord $_ -Target $instance -Continue
```

Comment:
So if it fails to grant one login you want it to continue failing?

Reply:
Yes. I've seen this behavior at other parts of this commandlet and in other commandlets.
But you are the expert and have the overview of the general structure in the commandlets.
So what do you suggest?

Reply:
Well since this one is actually trying to pull in the account, if it was not able to do that then continuing would cause additional errors (e.g. that `$InputObject` would be null/empty)...at least I would think it would.

I'm not sure if the test for this command is that extensive to cover it.

Reply:
No it wouldn't. If $InputObject would be $null then the foreach ($account in $InputObject) would just do nothing. 
If passing in two logins via $Login, and the first fails to be created because of a typing error, do we want to create and grant the second or do we want the commandlet to fail?